### DISCUSSION TITLE
Add `lossless` prop to `next/image` component

### DISCUSSION THREAD

**Discussion by styfle** [vercel/next.js#64639] [Ideas]
### Describe the feature you'd like to request

Some images have poor quality from lossy compression when converting to webp or avif formats, even when `quality={100}`.

In particular, gradients suffer from [Color Banding](https://en.wikipedia.org/wiki/Color_banding).

Examples:

- https://steventey.com/blog
- https://twitter.com/adrianmg/status/1466906559565348866

### Describe the solution you'd like

This can be resolved with a new prop, `lossless={true}`, that can be used to enable lossless compression.

Its supported by many tools already such as:

- `squoosh`
- [sharp](https://sharp.pixelplumbing.com/api-output#avif)
- [imgix](https://docs.imgix.com/apis/rendering/format/lossless)

### Describe alternatives you've considered

An alternative is to use `q={100}` to indicate lossless which seems to be how [cloudinary](https://cloudinary.com/documentation/image_optimization#:~:text=Images%20in%20the%20WebP%20format%20are%20lossless%20when%20specifying%20a%20quality%20of%20100) does it.

This would be much easier to implement than adding a new prop, but might be a breaking change.

**Comment by adrianmg** [comment]
Some more information: https://github.com/vercel/next.js/discussions/28614#discussioncomment-1747691

I'm not sure what's being used behind the scenes but `cwebp` offers two similar flags `-lossless` and `near_lossless <int> (0-100)` to adjust things as needed. Having one of them will probably be enough to avoid running into banding issues 👍

**Comment by Brooooooklyn** [comment]
Parcel has an lossless image optimizer based on `mozjpeg` and `oxipng`: https://github.com/parcel-bundler/parcel/blob/v2/packages/optimizers/image/src/lib.rs

Ref: https://twitter.com/devongovett/status/1439700949551906816

**Comment by predaytor** [comment]
Any updates? Using the Astro framework with Vercel Image Optimization (`/_vercel?image=`). Images regardless of the format (avif, webp) and quality (90-100) look desaturated and have the artifacts described above. This is unacceptable. What's the point of small image sizes if they look terrible? They need to do something about compression.

**Comment by aaronpurewal** [comment]
bump

**Comment by trieu-hphuc96** [comment]
any updates on this feature?
